### DISCUSSION TITLE
Websocket Routes Only Work on FastAPI, not APIRouter

### DISCUSSION THREAD

**Discussion by azurelotus06** [fastapi/fastapi#11779] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()

ws_router = APIRouter()
@ws_router.websocket("/ws")
async def websocket_endpoint(websocket: WebSocket):
    await manager.connect(websocket)

    while websocket.client_state.CONNECTED:
        try:
            await websocket.send_json({"Hello": "World!"})
            time.sleep(3)
        except (WebSocketDisconnect, ConnectionClosed):
            await manager.disconnect(websocket)
            break

file_router = APIRouter()
@filerouter.post("/fetch-all",
             response_model=FetchItemResponse)
async def fetch_all_items(
        credentials: Annotated[HTTPAuthorizationCredentials, Depends(security)],
        userid: Annotated[str, Form()],
        upload_repo: UploadRepository = Depends(get_mongodb_repo(UploadRepository))
    ):
    try:
        print("userid", userid)
        item_list, total = upload_repo.list_items(userid=userid)
        print(item_list)
        return FetchItemResponse(itemlist=[FetchModel(id=item.id,originname=item.originname, filepath=item.filepath, createdat=item.createdat, status=item.status) for item in item_list])
    except Exception as e:
        print(e)
        return None
app.include_router(file_router, prefix="/files", tags=["files"])
app.include_router(ws_router, prefix="/ws", tags=["websocket"])
```

### Description

**Similarity**
The same issue as this closed issue. (https://github.com/tiangolo/fastapi/issues/98)
It was closed because an owner declared that this issue was fixed. But I have same issue now with latest fastapi version and python stable version.
**Describe the bug**
I developed a full-stack project using fastapi, websocket and Next. I can fetch all file items when I include file_router only, but it fails when I include both routers. When I trace the request in Network tab in Google Inspect panel, I find its status is pending and has no response. Nothing happens in fastapi server, whether or not I send post request.

### Operating System

Windows

### FastAPI Version

0.111.0

### Pydantic Version

2.7.4

### Python Version

3.12.4

**Comment by YuriiMotov** [comment]
You are using `time.sleep(3)` inside of `async` function (websocket endpoint). It slows down your app. I guess this is probably the root of your issue

**Reply by azurelotus06** [reply]
I'm sorry but it's the same when I delete the line.

**Reply by YuriiMotov** [reply, marked as answer]
Try changing `time.sleep(3)` to `await asyncio.sleep(3)`.
`time.sleep` is blocking function, in async functions you should use `await asyncio.sleep` instead.

If this doesn't help, you probably have errors in the implementation of `manager`.

**Reply by azurelotus06** [reply]
Thanks, it works.
